lib/redmine_zulip/api.rb

```diff
@@ -0,0 +1,47 @@
+module RedmineZulip
+  class Api
+    attr_reader :url, :email, :key
+
+    def initialize
+      @url = Setting.plugin_redmine_zulip["zulip_url"]
+      @key = Setting.plugin_redmine_zulip["zulip_api_key"]
+      @email = Setting.plugin_redmine_zulip["zulip_email"]
+    end
+
+    def configured?
+      url.present? && email.present? && key.present?
+    end
+
+    def messages
+      RedmineZulip::Api::Messages.new(self)
+    end
+
+    private
+
+    class Messages
+      attr_reader :api
+
+      def initialize(api)
+        @api = api
+      end
+
+      def send(type:, to:, content:, subject: nil)
+        form_data = {
+          "type" => type,
+          "to" => to,
+          "content" => content
+        }
+        form_data["subject"] = subject if subject.present?
+        uri = URI("#{api.url}/v1/messages")
+        req = Net::HTTP::Post.new(uri)
+        req.basic_auth(api.email, api.key)
+        req["User-Agent"] = "RedmineZulip/#{RedmineZulip::VERSION}"
```

Comment:
Let's call this `ZulipRedminePlugin/${version...}`; that's our standard naming convention for plugins (putting the `Zulip` first).